### PR TITLE
[Impl 1] Equihash clean-up and tests

### PR DESCRIPTION
**Pull request by AlexandraRoatis**
## Description

Equihash classes clean-up based on IntelliJ's code inspection and improved test coverage.

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [ ] Bug fix.
- [ ] New feature.
- [x] Enhancement.
- [x] Unit test.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

Please describe the tests you used to validate this pull request. Provide any relevant details for test configurations as well as any instructions to reproduce these results.

- existing test suite and newly added tests

## Verification

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] I have self-reviewed my own code and conformed to the style guidelines of this project.
- [x] New and existing tests pass locally with my changes.
- [x] I have added tests for my fix or feature.
- [ ] I have made appropriate changes to the corresponding documentation.
- [x] My code generates no new warnings.
- [ ] Any dependent changes have been made.

### CODE REVIEW

File: modAionImpl/src/org/aion/equihash/EquiUtils.java [hunk 1]
```diff
@@ -64,11 +59,14 @@ else if (out == null)
...
-                    // Apply bit_len_mask across byte boundaries
-                    (bitLenMask >>> (8 * (outWidth - x - 1))) & 0xFF));
+                    out[j + x] =
+                            (byte)
+                                    ((
+                                            // Big-endian
+                                            accValue >>> (accBits + (8 * (outWidth - x - 1))))
+                                            & (
+                                            // Apply bit_len_mask across byte boundaries
+                                            (bitLenMask >>> (8 * (outWidth - x - 1))) & 0xFF));
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/equihash/EquiUtils.java]: can we format the code better?

File: modAionImpl/src/org/aion/equihash/EquiUtils.java [hunk 2]
```diff
@@ -180,8 +170,12 @@ else if (out == null)
...
                 accValue = accValue << bitLen;
                 for (int x = bytePad; x < inWidth; x++) {
-                    accValue = accValue | ((in[j + x] & ((bitLenMask >> (8 * (inWidth - x - 1))) & 0xFF)) << (8
-                            * (inWidth - x - 1)));
+                    accValue =
+                            accValue
+                                    | ((in[j + x]
+                                                    & ((bitLenMask >> (8 * (inWidth - x - 1)))
+                                                            & 0xFF))
+                                            << (8 * (inWidth - x - 1)));
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/equihash/EquiUtils.java]: same here, the formatter

File: modAionImpl/src/org/aion/equihash/OptimizedEquiValidator.java [hunk 3]
```diff
@@ -58,23 +51,23 @@ public OptimizedEquiValidator(int n, int k) {
         this.indicesHashLength = (n + 7) / 8;
         this.hashOutput = indicesPerHashOutput * indicesHashLength;
         this.collisionBitLength = n / (k + 1);
-        this.collisionByteLength = (collisionBitLength + 7) / 8;
+        // int collisionByteLength = (collisionBitLength + 7) / 8;
```
Comment (aionick) [inline modAionImpl/src/org/aion/equihash/OptimizedEquiValidator.java]: were these commented-out pieces meant to be deleted?

File: modAionImpl/src/org/aion/equihash/OptimizedEquiValidator.java [hunk 4]
```diff
@@ -179,53 +179,64 @@ private void genHash(byte[] blockHeader, byte[] nonce, Blake2b blake, int[] indi
 
         byte[] tmpHash = blake.digest();
 
-        System.arraycopy(tmpHash, (indices[index] % indicesPerHashOutput) * indicesHashLength, hash, 0, indicesHashLength);
+        System.arraycopy(
```
Comment (aionick) [inline modAionImpl/src/org/aion/equihash/OptimizedEquiValidator.java]: formatting here

File: modAionImpl/src/org/aion/equihash/OptimizedEquiValidator.java (hunk 3 above)
Comment (AlexandraRoatis) [reply modAionImpl/src/org/aion/equihash/OptimizedEquiValidator.java]: IntelliJ's code inspection pointed out that they were not used. They may become useful in the future, which is why I only commented the code.

File: modAionImpl/src/org/aion/equihash/OptimizedEquiValidator.java (hunk 4 above)
Comment (AlexandraRoatis) [reply modAionImpl/src/org/aion/equihash/OptimizedEquiValidator.java]: it was automatic, but I actually find this more readable

File: modAionImpl/src/org/aion/equihash/OptimizedEquiValidator.java [hunk 5]
```diff
@@ -179,53 +179,64 @@ private void genHash(byte[] blockHeader, byte[] nonce, Blake2b blake, int[] indi
...
 
-        for(int i = 0; i < indicesHashLength; i++)
-            hash[i] = (byte)(hash0[i] ^ hash1[i]);
+        for (int i = 0; i < indicesHashLength; i++) hash[i] = (byte) (hash0[i] ^ hash1[i]);
 
         int bits = (round < k ? round * collisionBitLength : n);
 
-        for(int i = 0; i < bits/8; i++) {
-            if(hash[i] != 0) {
+        for (int i = 0; i < bits / 8; i++) {
```
[the remark this answers is not in the document]
Comment (AionJayT) [reply modAionImpl/src/org/aion/equihash/OptimizedEquiValidator.java]: cause it has been computed twice

**Pull request merged**